File: sdks/python/bolt_prediction_sdk/models/hi_carbon_dual_view_response.py
"""
HICarbonDualViewResponse 模型
"""

from typing import Optional, Any, Dict, List
from datetime import datetime
from pydantic import Field

from .base import SDKBaseModel


class HiCarbonDualViewResponse(SDKBaseModel):
    """HI rollup 与碳排并列展示响应"""

    report_month: str = Field(description="报告月份 YYYY-MM")
    total_nodes: int = Field(description="节点总数")
    items: List[HiCarbonDualItemSchema] = Field(description="HI与碳排并列数据列表")
    generated_at: datetime = Field(description="生成时间")

File: sdks/java/src/main/java/com/boltprediction/sdk/model/HiCarbonDualViewResponse.java
package com.boltprediction.sdk.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import java.util.List;
import java.util.Map;
import java.time.OffsetDateTime;

/** HI rollup 与碳排并列展示响应 */
@JsonIgnoreProperties(ignoreUnknown = true)
public class HiCarbonDualViewResponse {

    @JsonProperty("report_month")
    private String reportMonth;

    @JsonProperty("total_nodes")
    private Integer totalNodes;

    @JsonProperty("items")
    private List<HiCarbonDualItemSchema> items;

    @JsonProperty("generated_at")
    private OffsetDateTime generatedAt;

    public String getReportMonth() {
        return reportMonth;
    }

    public void setReportMonth(String reportMonth) {
        this.reportMonth = reportMonth;
    }

    public Integer getTotalNodes() {
        return totalNodes;
    }

    public void setTotalNodes(Integer totalNodes) {
        this.totalNodes = totalNodes;
    }

    public List<HiCarbonDualItemSchema> getItems() {
        return items;
    }

    public void setItems(List<HiCarbonDualItemSchema> items) {
        this.items = items;
    }

    public OffsetDateTime getGeneratedAt() {
        return generatedAt;
    }

    public void setGeneratedAt(OffsetDateTime generatedAt) {
        this.generatedAt = generatedAt;
    }

}